test/built-ins/Map/prototype/forEach/iterates-values-deleted-then-readded.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
```

Comment:
This test is redundant with test/built-ins/Map/prototype/forEach/deleted-values-during-foreach.js . I suggest keeping them both but updating the latter to *only* delete a value